### ISSUE TITLE
Another bug in seqManager and performence increase

### ISSUE THREAD

**Issue by penguinol** [versatica/mediasoup#395]
Hi, guys!
There seem to be another bug in seqManager.
See such case:
```
	SECTION("sync and drop some input near max-value")
	{
		// clang-format off
		std::vector<TestSeqManagerInput<uint16_t>> inputs =
		{
			{ 65530,  1,  true, false },
			{ 65531,  2, false, false },
			{ 65532,  3, false, false },
			{ 65533,  0, false, true  },
			{ 65534,  0, false, true  },
			{ 65535,  4, false, false },
			{     0,  5, false, false },
			{     1,  6, false, false },
			{     2,  7, false, false },
			{     3,  8, false, false }	
		};
		// clang-format on

		SeqManager<uint16_t> seqManager;
		validate(seqManager, inputs);
	}
```

When we droped some input near the max-value, after input back to 0, the output secquence num will be discoutinued.

I think it's because we simple use `<` to compare the input with the dropped packets, as follow:
in `bool SeqManager<T>::Input(const T input, T& output)`
```
size_t dropped = std::count_if(
			  this->dropped.begin(), this->dropped.end(), [&input](T i) { return i < input; });
```
In this case, in fact, 65535 is smaller than 0 in sequence num, but larger in mathematics.
So we should use `IsSeqLowerThan` instead.

And also, i found that the performence of `count_if` is really low, here is a flame graph when about 20 peers in a conference, cpu is near 100%.
![perf](https://user-images.githubusercontent.com/14129614/81541230-6022c600-93a5-11ea-8320-c87b8d3cf295.png)
`count_if` takes more than 60% cpu time.
I think it's because `count_if` simply travel the map and find out how many items match the condition. When there are a lot of dropped packets, it will cost a lot of time.
Since we should use `IsSeqLowerThan` instead of `<`,and  `IsSeqLowerThan` is the defalut compare function of dropped map, we can just use `this->dropped.lower_bound(input)` to find the iterator, and use `std::distance` to calculate the number of dropped packets which are "smaller" than input.
Futhermore, in most of case, the number of dropped packets which are larger than input is far more less than those are smaller.
So `this->dropped.size() - std::distance(this->dropped.upper_bound(input), this->dropped.end())` might be faster than using ` std::distance(this->dropped.begin(), this->dropped.upper_bound(input))`, because `map::iterator` is not a random access iterator.

**Comment by ibc** [comment]
Amazing catch!

Would you mind creating a PR with the corresponding unit test, plea.... oh, you did!

**Comment by jmillan** [comment]
Great points @penguinol!

**Comment by jmillan** [comment]
@penguinol, can you share how do you get that flame graph?

**Comment by ibc** [comment]
http://www.brendangregg.com/FlameGraphs/cpuflamegraphs.html

**Comment by penguinol** [comment]
> @penguinol, can you share how do you get that flame graph?

```
perf record -F 1000 -g -p {pid}
perf script -i perf.data &> perf.unfold
FlameGraph-master/stackcollapse-perf.pl perf.unfold &> perf.folded
FlameGraph-master/flamegraph.pl perf.folded > perf.svg
```

**Comment by ibc** [comment]
Fixed in 3.5.11 thanks to PR #396 by @penguinol. Thanks!
